runtime/lua/vim/lsp.lua

```diff
@@ -928,10 +930,13 @@ end
 ---               Can also be passed as a function that returns the params table for cases where
 ---               parameters are specific to the client.
 ---@param handler lsp.MultiHandler (function)
---- Handler called after all requests are completed. Server results are passed as
---- a `client_id:result` map.
+---       Handler called after all requests are completed. Server results are passed as
+---       a `client_id:result` map.
+---@param on_unsupported? fun()
```

Comment:
Can we avoid doing this and just use `get_clients` to check for unsupported before calling `buf_request_all`?

Reply:
Also note, many of the functions in `lsp/buf.lua` don't use `buf_request_all` and simply call all the clients manually:

```lua
function M.references(context, opts)
  validate('context', context, 'table', true)
  local bufnr = api.nvim_get_current_buf()
  local clients = lsp.get_clients({ method = ms.textDocument_references, bufnr = bufnr })
  if not next(clients) then
    return
  end
  local win = api.nvim_get_current_win()

  local all_items = {}

  local function on_done()
    if not next(all_items) then
      vim.notify('No references found')
    else
      ...
    end
  end

  local remaining = #clients
  for _, client in ipairs(clients) do
    local params = ...
    client:request(ms.textDocument_references, params, function(_, result)
      local items = util.locations_to_items(result or {}, client.offset_encoding)
      vim.list_extend(all_items, items)
      remaining = remaining - 1
      if remaining == 0 then
        on_done()
      end
    end)
  end
end
```

So pick your poison. Eventually I plan to refactor this pattern out anyway,  so don't overthink it, just avoid extending any API.

Reply:
Okay, sorry I don't know about these plans.

Reply:
Yes `vim.lsp.buf_request_all` and `lsp.MultiHandler` will remain, but any changes to them is TBD.